frontera/contrib/backends/cassandra/components.py

```diff
@@ -0,0 +1,359 @@
+# -*- coding: utf-8 -*-
+import logging
+from datetime import datetime
+from time import time, sleep
+from cachetools import LRUCache
+from frontera.contrib.backends.partitioners import Crc32NamePartitioner
+from frontera.contrib.backends.memory import MemoryStates
+from frontera.core.components import Metadata as BaseMetadata, Queue as BaseQueue
+from frontera.core.models import Request, Response
+from frontera.utils.misc import get_crc32, chunks
+from frontera.utils.url import parse_domain_from_url_fast
+from cassandra.concurrent import execute_concurrent_with_args
+from frontera.contrib.backends.cassandra.models import Meta
+
+
+def retry_and_rollback(func):
+    def func_wrapper(self, *args, **kwargs):
+        tries = 5
+        while True:
+            try:
+                return func(self, *args, **kwargs)
+            except Exception, exc:
+                self.logger.exception(exc)
+                sleep(5)
+                tries -= 1
+                if tries > 0:
+                    self.logger.info("Tries left %i", tries)
+                    continue
+                else:
+                    raise exc
+    return func_wrapper
+
+
+class Metadata(BaseMetadata):
+    def __init__(self, session_cls, model_cls, cache_size, crawl_id='default'):
+        self.session = session_cls
+        self.model = model_cls
+        self.table = 'MetadataModel'
+        self.cache = LRUCache(cache_size)
+        self.logger = logging.getLogger("frontera.contrib.backends.cassandra.components.Metadata")
+        self.crawl_id = crawl_id
+
+    def frontier_stop(self):
+        pass
+
+    @retry_and_rollback
```

Comment:
Hi Alex,
first thanks for all the Fedback. 
No i've never triggered the retry logic.
Also I haven't seen the documentation about retry at the python driver. I try to switch to the cassandra driver retry.